api/vm/compile.ts

```diff
@@ -71,15 +71,14 @@ export function listenForCompileRequest() {
     try {
       const compiledResult = compile(compileConfig);
       const writer = new net.Socket({ fd: 4 });
-      writer.write(compiledResult);
+      writer.write(compiledResult, () => process.exit());
```

Comment:
i think we should avoid this, it feels like a code smell. just `await` `promisify`?

Reply:
Promisify doesn't seem to be valid on `writer.write`... or in any case I couldn't get it to work. How about this alternative?